### PR TITLE
feat: Enrolled Web Based Programs Implementation 

### PR DESCRIPTION
**Pull request by saeedbashir**
This PR adds support for web-based enrolled programs.

[[iOS] Program Progress Page](https://github.com/openedx/openedx-app-ios/issues/163)

https://github.com/openedx/openedx-app-ios/assets/5606473/a3af6a5d-23af-4469-b9e7-0accf414546f

### CODE REVIEW

File: Core/Core/Configuration/Config/Config.swift [hunk 1]
```diff
@@ -25,6 +25,8 @@ public protocol ConfigProtocol {
     var theme: ThemeConfig { get }
     var uiComponents: UIComponentsConfig { get }
     var discovery: DiscoveryConfig { get }
+    var program: DiscoveryConfig { get }
```
Comment (rnr) [inline Core/Core/Configuration/Config/Config.swift:28]: @saeedbashir  If Discovery Config now contains both Discovery and Program config, perhaps we need to rename it to some "level up" entry. Thoughts?

**Comment by rnr** [comment]
@saeedbashir  I know it's webView based content but perhaps we can add a small css injection to add some left padding?
Screenshot 2024-01-26 at 18 58 48

**Comment by rnr** [comment]
Also I would like to have a small css-injection here too
like add for "number-circle" class:
```css
display: table-cell;
vertical-align: middle;
```
Screenshot 2024-01-26 at 18 54 32
@saeedbashir  WDYT?

**Comment by rnr** [comment]
@saeedbashir  please look on this - looks like it publishes from background thread
Screenshot 2024-01-26 at 19 18 10

**Comment by saeedbashir** [comment]
Closing this PR in favor of https://github.com/openedx/openedx-app-ios/pull/260 because the PR was created directly from the openedX repo instead of fork. Thank you, @rnr for highlighting it.

**Pull request closed without merging**